### ISSUE TITLE
Unable to load DLL 'Microsoft.CognitiveServices.Speech.core.dll

### ISSUE THREAD

**Issue by karthikreddy408** [Azure-Samples/cognitive-services-speech-sdk#780]
When i try to implement the same sdk in my solution, I'm getting Unable to load DLL 'Microsoft.CognitiveServices.Speech.core.dll error. I have installed latest nuget version i.e. 1.13
If I Manually add this to my solution, It is throwing exception that i cannot add this.

Below are my solution Details.
Framework version: 4.6.2
Type: Windows desktop application
Nuget pkg version:1.13

**Comment by pankopon** [comment]
Hi,

First ensure that you have the latest "Microsoft Visual C++ Redistributable for Visual Studio 2015, 2017 and 2019" as the Speech SDK versions 1.13 and later depend on VS2019 runtimes. Download and install it from https://support.microsoft.com/en-us/help/2977003/the-latest-supported-visual-c-downloads.

Another thing is that depending on the Visual Studio version, there are apparently problems in handling packages.config (that the SDK examples use) correctly. The following was tested with Visual Studio 2019 version 16.7.2 (the latest as of this writing), following the instructions from https://docs.microsoft.com/en-us/nuget/consume-packages/migrate-packages-config-to-package-reference:
1. In Solution Explorer, right-click on packages.config and select "Migrate packages.config to PackageReference...."
2. In the opening dialog, there should be Microsoft.CognitiveServices.Speech in Top-level dependencies by default and no package compatibility issues.
3. Select OK to begin the migration.
4. After the migration is completed, rebuild the solution which should then run without problems.

With the above I can build and run e.g. https://github.com/Azure-Samples/cognitive-services-speech-sdk/tree/master/quickstart/csharp/dotnet/from-file successfully. Please let us know if this solves your issue. We will update the SDK examples to use PackageReference instead of packages.config hopefully already in the next release.

**Comment by karthikreddy408** [comment]
I tried with the above suggestions but still the issues is reoccurring.

**Comment by karthikreddy408** [comment]
Also other observation what i have found is, in my solution when active solution configuration is set to "Any CPU", it is throwing exception that i need to use X64 or X86 but in the sample application when the active solution configuration is set to Any CPU it is working fine.

**Comment by pankopon** [comment]
That sounds very odd. What version of Visual Studio are you using? We definitely recommend setting the active solution platform to a specific value in any case.

So if you still get the same error,
- Can you build and run SDK examples like https://github.com/Azure-Samples/cognitive-services-speech-sdk/tree/master/quickstart/csharp/dotnet/from-file successfully?
  - If yes, can you recreate your solution using such working example as a base?
- Does it not help if you clean your current solution and rebuild?
- Does it make any difference if you switch from Debug to Release configuration or vice versa, build and run?

If everything else fails, could you zip your solution folder (feel free to reduce the code to minimum) and attach it so that we can take a look.

**Comment by pankopon** [comment]
Closing the issue since no further updates, please open a new issue with details if more support is needed.

**Comment by amiddles** [comment]
I'd like to indicate that I've seen both issues now.
The dll files are in the published solution and still it does not seem to recognize them.
I'm building my project with Microsoft Visual Studio Professional 2022 (64-bit) - Current Version 17.2.5
